### PR TITLE
OpenSUSE Leap support in teuthology

### PR DESCRIPTION
**Pull request by rjfd**

### CODE REVIEW

**Comment by smithfarm** [comment]
@zmc OK for me to test this in the Sepia lab?

**Comment by zmc** [comment]
@smithfarm I don't have a problem with that; the only concern I'd have would be that we don't actually have OpenSUSE machines in the lab. Were you going to use vps hosts, or something else?

**Comment by smithfarm** [comment]
@zmc The openSUSE machines currently on work with teuthology-openstack. The purpose of the test in Sepia would be to demonstrate that the PR does not introduce any regressions. Sorry I should have made that clearer.

**Comment by zmc** [comment]
@smithfarm no problem, test away!

File: teuthology/task/install.py [hunk 1]
```diff
@@ -225,24 +225,50 @@ def _update_rpm_package_list_and_install(ctx, remote, rpm, config):
...
+    else:
+        proj_release = '{proj}-release-{release}.{dist_release}.noarch'.format(
+            proj=project, release=RELEASE, dist_release=dist_release)
     rpm_name = "{rpm_nm}.rpm".format(rpm_nm=proj_release)
     base_url = "{start_of_url}/noarch/{rpm_name}".format(
         start_of_url=start_of_url, rpm_name=rpm_name)
-    remote.run(args=['sudo', 'yum', '-y', 'install', base_url])
+    if dist_release == 'opensuse':
+        remote.run(args=[
+            'sudo', 'zypper', '--n', 'install', '--capability', rpm_name
```
Comment (zmc) [inline teuthology/task/install.py]: The ``--n`` arg looks funny to me; I don't know if I've ever seen a long arg that is a single letter. Typo or unicorn? :)

File: teuthology/task/install.py [hunk 2]
```diff
@@ -225,24 +225,50 @@ def _update_rpm_package_list_and_install(ctx, remote, rpm, config):
...
+        remote.run(
+            args=[
+                'sudo', 'yum', 'clean', 'all',
+            ])
 
     ldir = _get_local_dir(config, remote)
+
+    if dist_release == 'opensuse':
+        pkg_mng_opts = '-n'
+        pkg_mng_cmd = 'zypper'
```
Comment (zmc) [inline teuthology/task/install.py]: Could you put ``pkg_mng_cmd`` before ``pkg_mng_opts`` please? Just for readability.

File: teuthology/task/install.py [hunk 3]
```diff
@@ -225,24 +225,50 @@ def _update_rpm_package_list_and_install(ctx, remote, rpm, config):
...
+        pkg_mng_subcommand_opts = '--capability'
+    else:
+        pkg_mng_opts = '-y'
+        pkg_mng_cmd = 'yum'
```
Comment (zmc) [inline teuthology/task/install.py]: ``pkg_mng_cmd`` first please

File: teuthology/task/install.py [hunk 4]
```diff
@@ -434,30 +466,51 @@ def _remove_rpm(ctx, config, remote, rpm):
         pkglist=", ".join(rpm)))
     gitbuilder = _get_gitbuilder_project(ctx, remote, config)
     dist_release = gitbuilder.dist_release
+
+    if dist_release == 'opensuse':
+        pkg_mng_opts = '-n'
+        pkg_mng_cmd = 'zypper'
```
Comment (zmc) [inline teuthology/task/install.py]: ``pkg_mng_cmd`` first please

File: teuthology/task/install.py [hunk 5]
```diff
@@ -434,30 +466,51 @@ def _remove_rpm(ctx, config, remote, rpm):
...
+        pkg_mng_subcommand_opts = '--capability'
+    else:
+        pkg_mng_opts = '-y'
+        pkg_mng_cmd = 'yum'
```
Comment (zmc) [inline teuthology/task/install.py]: ``pkg_mng_cmd`` first please

File: teuthology/task/install.py [hunk 6]
```diff
@@ -849,28 +904,50 @@ def _upgrade_rpm_packages(ctx, config, remote, pkgs):
...
+
+    if gitbuilder.dist_release != 'opensuse':
+        uri = gitbuilder.uri_reference
+        _yum_fix_repo_priority(remote, project, uri)
+        _yum_fix_repo_host(remote, project)
+        _yum_set_check_obsoletes(remote)
+
+    if gitbuilder.dist_release == 'opensuse':
+        pkg_mng_opts = '-a'
+        pkg_mng_cmd = 'zypper'
```
Comment (zmc) [inline teuthology/task/install.py]: ``pkg_mng_cmd`` first please

File: teuthology/task/install.py [hunk 7]
```diff
@@ -849,28 +904,50 @@ def _upgrade_rpm_packages(ctx, config, remote, pkgs):
...
+    else:
+        pkg_mng_opts = 'all'
+        pkg_mng_cmd = 'yum'
```
Comment (zmc) [inline teuthology/task/install.py]: ``pkg_mng_cmd`` first please

File: teuthology/task/install.py (hunk 1 above)
Comment (smithfarm) [inline teuthology/task/install.py]: good catch, @zmc - it should be `-n` or `--non-interactive`

**Comment by smithfarm** [comment]
@rjfd See https://github.com/dachary/teuthology/pull/14

**Comment by rjfd** [comment]
@zmc pushed the corrections

**Comment by rjfd** [comment]
@zmc tomorrow I'll check this https://github.com/ceph/teuthology/blob/master/teuthology/packaging.py#L56-L105 and make the required changes

**Comment by rjfd** [comment]
@zmc regarding `packaging.py`, the idea is to use the functions `install_package / remove_package`, in the install task?

**Comment by rjfd** [comment]
@zmc I opened another PR #957 for using the `install_package / remove_package` in the install task, since it tackles an issue orthogonal to this PR.
I guess this PR is ready to merge?

**Comment by zmc** [comment]
Thanks @rjfd!

**Pull request merged**